src/base/bittorrent/session.h

```diff
@@ -92,7 +92,8 @@ namespace BitTorrent
         {
             Default = 0,
             MMap = 1,
-            Posix = 2
+            Posix = 2,
+            ForcePreadPwrite = 3
```

Comment:
And yet, why it is "**force** pread/pwrite"? Is there any other "pread/pwrite" based type available that isn't "force" unlike this one?

Reply:
It would be in the future.

Reply:
> It would be in the future.

In general, you can't know anything about it. We need to name this type only based on its distinctive features (relative to current types).
But since in this case we know about the new IO backend being developed, which may be added as a separate IO type in the future, we could immediately take this into account when naming. But anyway, is the current type (added by this PR) different from the new one in such a way that we can describe this difference with the word "force"? I don't believe it's appropriate.
So I would just name it "Pread/Pwrite" ("PreadPwrite" internally) since this says enough about its features relative to the rest of the existing IO types, and think about the name of the new type when it becomes available.

Reply:
>But anyway, is the current type (added by this PR) different from the new one in such a way that we can describe this difference with the word "force"? I don't believe it's appropriate.

There are some minor differences between the two approaches. If they are the same, then arvidn should be able to get it done quickly, isn't? Currently, it is like driving an hybrid car with batteries disabled. It will run but it isn't in the best condition.
I would really like to have differentiate between these two and not impersonating the fallback mode as a native one.

Reply:
The point is how it is presented at the user level.
You're talking about a case where the battery on/off switch is available to the user of a hybrid car. But in this case, we are talking about the case when the car is assembled from the same parts as the hybrid, but the batteries are not mounted at all. What's the point of talking about it as a hybrid if it can only work in one single mode? This low-level information could only be of interest to a very advanced user in the same way as the information that different processors with different numbers of cores and cache size are actually created from the same crystals, physically having the same number of cores, cache size, etc.

P.S. perhaps using "Force pread/pwrite" name would be acceptable if the existing type was called like "Hybrid (Mmap, Pread/Pwrite)".

Reply:
I think we should resolve this. It is very frustrating to be blocked for such insignificant reason. People want that functionality, the naming should be changed later.

Reply:
> the naming could be changed later.

It's not a problem to change name in UI. But if this internal option name leaks into the release, then in order to change it, we will either have to maintain both names (old and new), or add an update code replacing it in the configuration file.
>I think we should resolve this.

Agree.
@Chocobo1 
My reasoning is still looks valid for me. I appeal you to think again about whether `ForcePreadPwrite` is really better than `PreadPwrite`. Answer the question, then what will you name the new option so that it reflects its difference from this one? If everything suits you, then I give in, you can do it your own way - your responsibility.

Reply:
PS. For me, the following names look acceptable (both having only one option and both):
1. PreadPwrite, PreadPwrite2 (simplest case)
2. PreadPwrite, AdvancedPreadPwrite
3. PreadPwrite, BufferedPreadPwrite
4. SimplePreadPwrite, PreadPwrite

Reply:
> This is related to the code and config file, right?

Yes.
As I mentioned above, it is not difficult to change the text in the UI if the original version is later found unsuitable.


> I don't mind the UI also use it, although it is a little strange.

I think "Simplified pread/pwrite" would be acceptable first approach (at least it much better than "Force pread/pwrite".

Reply:
Honestly, the word `Simplified` brings more associations with Simplified Chinese for me. 👀
Anyway, waiting for a final decision.

Reply:
`Simplified`, `Simple` I don't have strong opinion on one. You can just pick one.

Reply:
Ok, I hope we agree on it.